File: specification/web/resource-manager/Microsoft.DomainRegistration/stable/2022-03-01/examples-js/UpdateAppServiceDomain.js
const { WebSiteManagementClient } = require("@azure/arm-appservice");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Description for Creates or updates a domain.
 *
 * @summary Description for Creates or updates a domain.
 * x-ms-original-file: specification/web/resource-manager/Microsoft.DomainRegistration/stable/2022-03-01/examples/UpdateAppServiceDomain.json
 */
async function updateAppServiceDomain() {
  const subscriptionId = "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345";
  const resourceGroupName = "testrg123";
  const domainName = "example.com";
  const domain = {
    authCode: "exampleAuthCode",
    autoRenew: true,
    consent: {
      agreedAt: new Date("2021-09-10T19:30:53Z"),
      agreedBy: "192.0.2.1",
      agreementKeys: ["agreementKey1"],
    },
    contactAdmin: {
      addressMailing: {
        address1: "3400 State St",
        city: "Chicago",
        country: "United States",
        postalCode: "67098",
        state: "IL",
      },
      email: "admin@email.com",
      fax: "1-245-534-2242",
      jobTitle: "Admin",
      nameFirst: "John",
      nameLast: "Doe",
      nameMiddle: "",
      organization: "Microsoft Inc.",
      phone: "1-245-534-2242",
    },
    contactBilling: {
      addressMailing: {
        address1: "3400 State St",
        city: "Chicago",
        country: "United States",
        postalCode: "67098",
        state: "IL",
      },
      email: "billing@email.com",
      fax: "1-245-534-2242",
      jobTitle: "Billing",
      nameFirst: "John",
      nameLast: "Doe",
      nameMiddle: "",
      organization: "Microsoft Inc.",
      phone: "1-245-534-2242",
    },
    contactRegistrant: {
      addressMailing: {
        address1: "3400 State St",
        city: "Chicago",
        country: "United States",
        postalCode: "67098",
        state: "IL",
      },
      email: "registrant@email.com",
      fax: "1-245-534-2242",
      jobTitle: "Registrant",
      nameFirst: "John",
      nameLast: "Doe",
      nameMiddle: "",
      organization: "Microsoft Inc.",
      phone: "1-245-534-2242",
    },
    contactTech: {
      addressMailing: {
        address1: "3400 State St",
        city: "Chicago",
        country: "United States",
        postalCode: "67098",
        state: "IL",
      },
      email: "tech@email.com",
      fax: "1-245-534-2242",
      jobTitle: "Tech",
      nameFirst: "John",
      nameLast: "Doe",
      nameMiddle: "",
      organization: "Microsoft Inc.",
      phone: "1-245-534-2242",
    },
    dnsType: "DefaultDomainRegistrarDns",
    privacy: false,
  };
  const credential = new DefaultAzureCredential();
  const client = new WebSiteManagementClient(credential, subscriptionId);
  const result = await client.domains.update(resourceGroupName, domainName, domain);
  console.log(result);
}

updateAppServiceDomain().catch(console.error);

File: specification/web/resource-manager/Microsoft.DomainRegistration/stable/2022-03-01/examples-java/UpdateAppServiceDomain.java
import com.azure.core.util.Context;
import com.azure.resourcemanager.appservice.models.Address;
import com.azure.resourcemanager.appservice.models.Contact;
import com.azure.resourcemanager.appservice.models.DnsType;
import com.azure.resourcemanager.appservice.models.DomainPatchResource;
import com.azure.resourcemanager.appservice.models.DomainPurchaseConsent;
import java.time.OffsetDateTime;
import java.util.Arrays;

/** Samples for Domains Update. */
public final class Main {
    /*
     * x-ms-original-file: specification/web/resource-manager/Microsoft.DomainRegistration/stable/2022-03-01/examples/UpdateAppServiceDomain.json
     */
    /**
     * Sample code: Update App Service Domain.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void updateAppServiceDomain(com.azure.resourcemanager.AzureResourceManager azure) {
        azure
            .webApps()
            .manager()
            .serviceClient()
            .getDomains()
            .updateWithResponse(
                "testrg123",
                "example.com",
                new DomainPatchResource()
                    .withContactAdmin(
                        new Contact()
                            .withAddressMailing(
                                new Address()
                                    .withAddress1("3400 State St")
                                    .withCity("Chicago")
                                    .withCountry("United States")
                                    .withPostalCode("67098")
                                    .withState("IL"))
                            .withEmail("admin@email.com")
                            .withFax("1-245-534-2242")
                            .withJobTitle("Admin")
                            .withNameFirst("John")
                            .withNameLast("Doe")
                            .withNameMiddle("")
                            .withOrganization("Microsoft Inc.")
                            .withPhone("1-245-534-2242"))
                    .withContactBilling(
                        new Contact()
                            .withAddressMailing(
                                new Address()
                                    .withAddress1("3400 State St")
                                    .withCity("Chicago")
                                    .withCountry("United States")
                                    .withPostalCode("67098")
                                    .withState("IL"))
                            .withEmail("billing@email.com")
                            .withFax("1-245-534-2242")
                            .withJobTitle("Billing")
                            .withNameFirst("John")
                            .withNameLast("Doe")
                            .withNameMiddle("")
                            .withOrganization("Microsoft Inc.")
                            .withPhone("1-245-534-2242"))
                    .withContactRegistrant(
                        new Contact()
                            .withAddressMailing(
                                new Address()
                                    .withAddress1("3400 State St")
                                    .withCity("Chicago")
                                    .withCountry("United States")
                                    .withPostalCode("67098")
                                    .withState("IL"))
                            .withEmail("registrant@email.com")
                            .withFax("1-245-534-2242")
                            .withJobTitle("Registrant")
                            .withNameFirst("John")
                            .withNameLast("Doe")
                            .withNameMiddle("")
                            .withOrganization("Microsoft Inc.")
                            .withPhone("1-245-534-2242"))
                    .withContactTech(
                        new Contact()
                            .withAddressMailing(
                                new Address()
                                    .withAddress1("3400 State St")
                                    .withCity("Chicago")
                                    .withCountry("United States")
                                    .withPostalCode("67098")
                                    .withState("IL"))
                            .withEmail("tech@email.com")
                            .withFax("1-245-534-2242")
                            .withJobTitle("Tech")
                            .withNameFirst("John")
                            .withNameLast("Doe")
                            .withNameMiddle("")
                            .withOrganization("Microsoft Inc.")
                            .withPhone("1-245-534-2242"))
                    .withPrivacy(false)
                    .withAutoRenew(true)
                    .withConsent(
                        new DomainPurchaseConsent()
                            .withAgreementKeys(Arrays.asList("agreementKey1"))
                            .withAgreedBy("192.0.2.1")
                            .withAgreedAt(OffsetDateTime.parse("2021-09-10T19:30:53Z")))
                    .withDnsType(DnsType.DEFAULT_DOMAIN_REGISTRAR_DNS)
                    .withAuthCode("exampleAuthCode"),
                Context.NONE);
    }
}
